package_managers/apt_get/apt_get.bzl

```diff
@@ -14,60 +14,115 @@
 
 """ Rules that create an output script to install packages via apt-get."""
 
-CACHE_DIR = "/install"
+CACHE_DIR = "/tmp/install"
 
 ARCHIVE_DIR = "."
 
+load(
+    "//package_managers:package_manager_provider.bzl",
+    "package_manager_provider",
+)
+load(
+    "@io_bazel_rules_docker//skylib:filetype.bzl",
+    tar_filetype = "tar",
+)
+
+def _generate_download_commands(ctx):
+    commands = []
+    # Fetch Index
+    commands.append('apt-get update -y')
+    # Make partial dir
+    commands.append('mkdir -p {0}/{1}/partial'.format(CACHE_DIR, ARCHIVE_DIR))
+    install_command = ('apt-get install --no-install-recommends -y -q -o ' +
+                       'Dir::Cache="{0}" -o Dir::Cache::archives="{1}" {2} ' +
+                       '--download-only'
+                      ).format(
+                          CACHE_DIR,
+                          ARCHIVE_DIR,
+                          ' '.join(ctx.attr.packages)
+                      )
+    # Install command
+    commands.append(install_command)
+    # Tar command to only include all the *.deb files and ignore other
+    # directories placed in the cache dir.
+    tar_command = ('tar -cpf {output}.tar --directory {cache}/{archive} ' +
+                   '`cd {cache}/{archive} && ls *.deb`'
+                  ).format(
+                      output=ctx.attr.name,
+                      cache=CACHE_DIR,
+                      archive=ARCHIVE_DIR
+                  )
+    commands.append(tar_command)
+    return commands
+
+def _generate_install_commands(ctx, tar):
+    commands = []
+    commands.append('tar -xvf {output}'.format(output=tar))
```

Comment:
It might be easier to maintain if you switch this to a single template string or file. WDYT?

Reply:
Same for generate_download_commands.

Reply:
there is no way to read a file in bazel other than writing a tool which would.
I can make this template string in code if it looks cleaner.

Reply:
Yeah sorry, I meant to either use a template file on disk and use ctx.actions.expand_template or to use a template variable in code. I don't really care either way.

I think you don't need to really read it since it ends up as a file on disk anyway in ctx.actions.write.

Reply:
ah, i will have to read it to pass it as a list of commands to the provider.